NebulaNetwork/PacketProcessors/Factory/Entity/UpgradeEntityRequestProcessor.cs

```diff
@@ -31,17 +31,32 @@ public override void ProcessPacket(UpgradeEntityRequest packet, NebulaConnection
 
                 Quaternion qRot = new Quaternion(packet.rot.x, packet.rot.y, packet.rot.z, packet.rot.w);
                 Vector3 vPos = new Vector3(packet.pos.x, packet.pos.y, packet.pos.z);
-                EntityData[] ePool = planet.factory.entityPool;
 
-                for (int i = 1; i < planet.factory.entityCursor; i++)
+                
+
+                for (int i = 1; i < (packet.IsPrebuild ? planet.factory.prebuildCursor : planet.factory.entityCursor); i++)
```

Comment:
prefer setting this before the loop, otherwise you're switching on packet.IsPrebuild on every loop iteration

Reply:
Or better yet, why not have 2 different loops here?